dev/Expander/Expander.cpp

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "Expander.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+#include "ExpanderAutomationPeer.h"
+#include "Utils.h"
+#include "winnls.h"
+#include "ExpanderCollapsedEventArgs.h"
+#include "ExpanderExpandingEventArgs.h"
+
+static constexpr auto c_toggleButton = L"ExpanderHeader"sv;
+
+Expander::Expander()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_Expander);
+
+    SetDefaultStyleKey(this);
+}
+
+winrt::AutomationPeer Expander::OnCreateAutomationPeer()
+{
+    return winrt::make<ExpanderAutomationPeer>(*this);
+}
+
+void Expander::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected{ *this };
+    auto toggleButton = GetTemplateChildT<winrt::ToggleButton>(c_toggleButton, *this);
```

Comment:
looks like you only need FrameworkElement, so could we just try to cast to that ?

Reply:
We need the framework element automation peer, so I think this needs to stay.